tracer/src/Datadog.Trace/Configuration/DynamicConfigurationManager.cs

```diff
@@ -77,7 +77,7 @@ private static void OnConfigurationChanged(ConfigurationBuilder settings)
                 TraceEnabled = settings.WithKeys(ConfigurationKeys.TraceEnabled).AsBool(),
                 // RuntimeMetricsEnabled = settings.WithKeys(ConfigurationKeys.RuntimeMetricsEnabled).AsBool(),
                 // DataStreamsMonitoringEnabled = settings.WithKeys(ConfigurationKeys.DataStreamsMonitoring.Enabled).AsBool(),
-                SamplingRules = settings.WithKeys(ConfigurationKeys.CustomSamplingRules).AsString(),
+                SamplingRules = settings.WithKeys(ConfigurationKeys.CustomSamplingRules).GetAs<string>(null, null, s => s),
```

Comment:
It's too bad we don't have a strongly typed way to convert the JArray into a result, but stringifying it right now can solve the problem in the short-term so 👍🏼